src/pyk/ktool/kprove.py

```diff
@@ -360,17 +382,27 @@ def implies(self, antecedent: CTerm, consequent: CTerm, bind_consequent_variable
         consequent_kore = self.kast_to_kore(_consequent, GENERATED_TOP_CELL)
         _, kore_client = self.kore_rpc()
         result = kore_client.implies(antecedent_kore, consequent_kore)
+        if type(result.implication) is not Top:
+            _LOGGER.warning(
+                f'Received a non-trivial implication back from check implication endpoint: {result.implication}'
+            )
+        if result.predicate is not None and type(result.predicate) is not Top:
+            raise ValueError(
+                f'Received a non-trivial predicate back from check implication endpoint: {result.predicate}'
+            )
         if result.substitution is None:
             return None
         ml_subst = self.kore_to_kast(result.substitution)
+        if is_top(ml_subst):
+            return Subst({})
         subst_pattern = mlEquals(KVariable('###VAR'), KVariable('###TERM'))
         _subst: Dict[str, KInner] = {}
         for ml_pred in flatten_label('#And', ml_subst):
             m = subst_pattern.match(ml_pred)
             if m is not None and type(m['###VAR']) is KVariable:
                 _subst[m['###VAR'].name] = m['###TERM']
             else:
-                raise ValueError(f'Recieved back a non-substitution from implies endpoint: {ml_pred}')
+                raise ValueError(f'Received a non-substitution from implies endpoint: {ml_pred}')
```

Comment:
If this assumption is part of the contract, this can even be an `AssertionError` I think.

```suggestion
                raise AssertionError(f'Received a non-substitution from implies endpoint: {ml_pred}')
```

Reply:
Done.